src/rust/engine/Cargo.toml

```diff
@@ -2,7 +2,14 @@
 version = "0.0.1"
 name = "engine"
 authors = [ "Pants Build <pantsbuild@gmail.com>" ]
-build = "src/cffi_build.rs"
+
+# NB: We use the custom build script below, but omit the entry and count on Cargo's default
+# behavior of looking for `build.rs` in the package root and using it (see:
+# http://doc.crates.io/build-script.html). By allowing the default scan proper cargo build
+# incremntalism is engaged; without it is not and O(10s) is sent no-op recompiling for
```

Comment:
"incrementalism"

Shall I file a ticket about this one upstream? Or is this one close enough? https://github.com/rust-lang/cargo/issues/3076

Reply:
It may in fact explain what we're doing wrong. Gimme more time here.